src/timeline/InputBar.cpp

```diff
@@ -450,15 +450,67 @@ InputBar::generateRelations() const
 void
 InputBar::message(const QString &msg, MarkdownOverride useMarkdown, bool rainbowify)
 {
+    auto trimmed                = msg.trimmed();
     mtx::events::msg::Text text = {};
-    text.body                   = msg.trimmed().toStdString();
+    text.body                   = trimmed.toStdString();
+
+    // we don't want to have unexpected behavior if somebody was genuinely trying to edit a message
+    // to be a regex
+    if (UserSettings::instance()->sedEditing() && room->edit().isEmpty()) [[unlikely]] {
+        // The bulk of this logic was shamelessly stolen from Neochat. Thanks to Carl Schwan for
+        // letting us have this :)
+        static const QRegularExpression sed("^s/([^/]*)/([^/]*)(/g)?$");
+        auto match = sed.match(trimmed);
+
+        if (match.hasMatch()) {
+            const QString regex       = match.captured(1);
+            const QString replacement = match.captured(2).toHtmlEscaped();
+            const QString flags       = match.captured(3);
+
+            std::optional<mtx::events::collections::TimelineEvents> event;
+            if (!room->reply().isEmpty()) {
+                auto ev = room->eventById(room->reply());
+                if (ev && room->data(*ev, TimelineModel::IsEditable).toBool())
+                    event = ev;
+            } else {
+                for (int i = 0; i < room->rowCount(); ++i) {
+                    const auto idx = room->index(i);
+                    if (room->data(idx, TimelineModel::IsEditable).toBool()) {
+                        event = room->eventById(room->data(idx, TimelineModel::EventId).toString());
+                        break;
+                    }
+                }
+            }
+
+            if (event) {
+                auto other = room->data(*event, TimelineModel::FormattedBody).toString();
+                const auto original{other};
+                if (flags == "/g")
+                    other.replace(regex, replacement);
+                else
+                    other.replace(other.indexOf(regex), regex.size(), replacement);
```

Comment:
That way the first one will do \1 group replacement, the other one won't.

The proper solution is probably to extract the first match and only do a replace on that, then stitch it back together. The regex.size() also is wrong, since it will behave incorrectly with /s/.*// imo.

Reply:
I'm not sure that I understand what you are saying here. Could you elaborate?

Reply:
1. If you do a replacement with `s/.*//`, I think your code will just delete 2 characters at the start of the message.
2. If you do a replacement like `s/[abc]*/__\1__/` the first one will replace `aaacbbbaaa` with `__aaacbbbaaa__`, the second one will replace it with the literal text `__\1__`, I think.